packages/mobile/src/import/ImportContacts.tsx

```diff
@@ -0,0 +1,137 @@
+import Button, { BtnTypes } from '@celo/react-components/components/Button'
+import colors from '@celo/react-components/styles/colors'
+import { fontStyles } from '@celo/react-components/styles/fonts'
+import * as React from 'react'
+import { WithNamespaces, withNamespaces } from 'react-i18next'
+import { ScrollView, StyleSheet, Text, View } from 'react-native'
+import { connect } from 'react-redux'
+import { componentWithAnalytics } from 'src/analytics/wrapper'
+import DevSkipButton from 'src/components/DevSkipButton'
+import { Namespaces } from 'src/i18n'
+import VerifyAddressBook from 'src/icons/VerifyAddressBook'
+import { importContacts } from 'src/identity/actions'
+import { navigate } from 'src/navigator/NavigationService'
+import { Screens } from 'src/navigator/Screens'
+import DisconnectBanner from 'src/shared/DisconnectBanner'
+import { requestContactsPermission } from 'src/utils/androidPermissions'
+
+interface DispatchProps {
+  importContacts: typeof importContacts
+}
+
+type Props = WithNamespaces & DispatchProps
+
+class ImportContacts extends React.Component<Props> {
+  static navigationOptions = {
+    headerStyle: {
+      elevation: 0,
+    },
+    headerLeft: null,
+    headerRightContainerStyle: { paddingRight: 15 },
+    headerRight: (
+      <View>
+        <DisconnectBanner />
+      </View>
+    ),
+  }
+
+  nextScreen = () => {
+    navigate(Screens.VerifyEducation)
+  }
+
+  onPressEnable = async () => {
+    // TODO (Sally) Import contacts duplication
+    requestContactsPermission().then(async (response) => {
+      await this.props.importContacts()
```

Comment:
This does not do what you think it will.
this.props.importContacts() is synchronous. It just dispatches an event, which Saga picks up to trigger an async saga. using `await` here has no effect.
Let's remove it, make this callback sync, and we'll address the wait-here-til-imports-done feature in a follow-up PR.
Also, you seem to call importContacts here regardless of the response. We should probably only do it if we received permissions right?